### PR TITLE
Check for spelling typos in docs as part of CI/CD

### LINKED ISSUE #14623: Use codespell to catch spelling errors in docs as part of pre-commit
Codespell is part of pre-commit but docs are excluded:
```
 exclude: package-lock.json|_vendor/.*|docs/.*
```

### PR DESCRIPTION
**Pull request by discdiver**
Includes docs directory in codespell pre-commit hook.

Closes #14623.

Tested locally and passed.

### Checklist

- [X] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
- [X] This pull request references any related issue by including "closes `<link to issue>`"
  
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [ ] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,50 +1,50 @@
...
-        entry: ./scripts/generate_mintlify_openapi_docs.py
-        pass_filenames: false
-        files: |
-          (?x)^(
-              .pre-commit-config.yaml|
-              src/prefect/server/api/.*|
-              src/prefect/server/schemas/.*|
-              scripts/generate_mintlify_openapi_docs.py
-          )$
+    - repo: https://github.com/astral-sh/ruff-pre-commit
```
Comment (zzstoatzz) [inline .pre-commit-config.yaml:2]: the spacing in this diff looks pretty weird - was this intentional?

if not, any chance we can omit all the spacing changes?

Comment (discdiver) [reply .pre-commit-config.yaml:2]: yeah, not sure why that auto-formatted like that. Let me change it back.

Comment (discdiver) [reply .pre-commit-config.yaml]: Done.

File: docs/3.0rc/api-ref/rest-api/server/schema.json [hunk 2]
```diff
@@ -22279,6 +22279,11 @@
                         "title": "Prefect Task Scheduling Pending Task Timeout",
                         "default": "PT0S"
                     },
+                    "PREFECT_EXPERIMENTAL_ENABLE_EXTRA_RUNNER_ENDPOINTS": {
+                        "type": "boolean",
+                        "title": "Prefect Experimental Enable Extra Runner Endpoints",
+                        "default": false
+                    },
```
Comment (zzstoatzz) [inline docs/3.0rc/api-ref/rest-api/server/schema.json:22286]: i think you may want to remove this, as this appears to be why the statis analysis is failing

**Comment by zzstoatzz** [comment]
@discdiver there's still a failing pre-commit check [here](https://github.com/PrefectHQ/prefect/actions/runs/9960852656/job/27521033984?pr=14624#step:8:59), extra newline it seems?

**Review by zzstoatzz — APPROVED** [review]

**Pull request merged**